packages/web-components/src/components/va-promo-banner/va-promo-banner.tsx

```diff
@@ -0,0 +1,170 @@
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * anchor link or the dismiss icon is clicked and disableAnalytics is not true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  /**
+   * Keep track of locally dismissed Banners
+   * */
+  @State() dismissedBanners: Array<String> = [];
+
+  private closeHandler(): void {
+    this.closeEvent.emit();
+    // Derive the updated dismissed banners.
+    const updatedDismissedBanners = [...this.dismissedBanners, this.bannerId];
+
+    // Update dismissedBanners in state.
+    this.dismissedBanners = updatedDismissedBanners;
+
+    // Add the promo banner ID to local storage.
+    localStorage.setItem(
+      DISMISSED_PROMO_BANNERS_LOCAL_STORAGE_KEY,
+      JSON.stringify(updatedDismissedBanners),
+    );
+  }
+
+  private handleLinkClick(e: MouseEvent): void {
+    if (!this.disableAnalytics) {
+      const targetEl = e.target as HTMLElement;
+      const targetAnchor =
+        targetEl?.tagName === 'A' ||
+        (targetEl?.tagName === 'SLOT' &&
+          targetEl?.parentElement.tagName === 'A');
+      // If it's a link being clicked, dispatch an analytics event
+      if (targetAnchor) {
+        const detail = {
+          componentName: 'va-promo-banner',
+          action: 'linkClick',
+          details: {
+            text:
+              targetEl.tagName === 'A' ? targetEl.innerText : this.el.innerText,
+            href: this.href,
+            target: this.target,
+            type: this.type,
+          },
+        };
+        this.componentLibraryAnalytics.emit(detail);
+      }
+    }
+  }
+
+  componentWillLoad() {
+    // Derive dismissed banners from storage.
+    const dismissedBannersString = localStorage.getItem(
+      DISMISSED_PROMO_BANNERS_LOCAL_STORAGE_KEY,
+    );
+    this.dismissedBanners = dismissedBannersString
+      ? JSON.parse(dismissedBannersString)
+      : [];
+  }
+
+  render() {
+    // Derive if the banner is dismissed.
+    const isBannerDismissed = this.dismissedBanners?.includes(this.bannerId);
+
+    // Do not render if the promo banner is dismissed.
+    if (isBannerDismissed) {
+      return null;
+    }
+
+    return (
+      <Host>
+        <div class="va-banner-body">
+          <div class="va-banner-icon">
+            <div class="va-banner-icon-contents">
+              <i aria-hidden="true" class={this.type} role="presentation" />
+            </div>
+          </div>
+          <div class="va-banner-content">
+            {this.renderCustom ? (
+              <slot></slot>
+            ) : (
+              <a
+                class="va-banner-content-link"
+                href={this.href}
+                target={this.target}
+                onClick={e => this.handleLinkClick(e)}
+              >
+                <slot></slot> <i aria-hidden="true" role="presentation" />
```

Comment:
Is passing `<Link>` a possible use case here? Will it cause a page reload?

Reply:
Do we have an example of using the `<Link>` tag in another Web Component? Thought that was a React Router thing

Reply:
After a second look at this, devs can use `renderCustom` if they need to use `<Link>`.

Here's an example for my use case, but I don't think it's useful here. I needed to wrap children with a `li` element and tried this approach, but it caused page reloads when using React Router's `<Link>`.
https://github.com/department-of-veterans-affairs/component-library/blob/69655465824e556cc6ea5291be94557e61448986/packages/web-components/src/components/va-breadcrumbs/va-breadcrumbs.tsx#L70-L91

This was the solution and accomplished the same thing without transforming the node.
https://github.com/department-of-veterans-affairs/component-library/blob/c145ac3802840c39f0b6e94b14c94146829a3a90/packages/web-components/src/components/va-breadcrumbs/va-breadcrumbs.tsx#L67-L78

Reply:
Yeah if they are going to do something atypical like that render-custom should be the prop to use

Reply:
Awesome, thank you for that information.